packages/safe-ds-vscode/src/extension/EDAPanel.ts

```diff
@@ -0,0 +1,325 @@
+  public static context: vscode.ExtensionContext;
+
+  public static readonly viewType = "eda";
+
+  private readonly _panel: vscode.WebviewPanel;
+  private readonly _extensionUri: vscode.Uri;
+  private _disposables: vscode.Disposable[] = [];
+  private _tableIdentifier: string | undefined;
+  private _pythonServerPort: number = 5000;
+  private _startPipelineId: string = "";
+  private _column: vscode.ViewColumn | undefined;
+  private _webviewListener: vscode.Disposable | undefined;
+  private _viewStateChangeListener: vscode.Disposable | undefined;
+  // private _lastVisibleState: boolean = true;
+
+  private constructor(panel: vscode.WebviewPanel, extensionUri: vscode.Uri, startPipeLineId: string, tableIdentifier?: string, pythonServerPort = 5000) {
+    this._panel = panel;
+    this._extensionUri = extensionUri;
+    this._tableIdentifier = tableIdentifier;
+    this._pythonServerPort = pythonServerPort;
+    this._startPipelineId = startPipeLineId;
+
+    // Set the webview's initial html content
+    this._update();
+
+    // Listen for when the panel is disposed
+    // This happens when the user closes the panel or when the panel is closed programatically
+    this._panel.onDidDispose(() => this.dispose(), null, this._disposables);
+
+    // Handle view state changes
+    this._viewStateChangeListener = this._panel.onDidChangeViewState(async (e) => {
+      const updatedPanel = e.webviewPanel; 
+      if (updatedPanel.visible) {
+        this._column = updatedPanel.viewColumn;
+        // Not needed as retainContextWhenHidden is true for panel
+        // if(!this._lastVisibleState) { // State only has to be updated if it left the visible state and thus lost it's state
+        //   webviewApi.postMessage(updatedPanel.webview, {
+        //     command: "setWebviewState",
+        //     value: await this.constructCurrentState(),
+        //   });
+        // }
+      }
+      // this._lastVisibleState = e.webviewPanel.visible;
+    });
+    this._disposables.push(this._viewStateChangeListener);
+
+    // Handle messages from the webview
+    const webview = this._panel.webview;
+    this._webviewListener = webview.onDidReceiveMessage(async (data: ToExtensionMessage) => {
+      console.log(data.command + " called")
+      switch (data.command) {
+        case "setInfo": {
+          if (!data.value) {
+            return;
+          }
+          vscode.window.showInformationMessage(data.value);
+          break;
+        }
+        case "setError": {
+          if (!data.value) {
+            return;
+          }
+          vscode.window.showErrorMessage(data.value);
+          break;
+        }
+        case "setCurrentGloabalState": {
+          if (!data.value) {
+            return;
+          }
+          const existingStates = EDAPanel.context.globalState.get("webviewState") as State[];
+          const stateExists = existingStates.some(s => s.tableIdentifier === data.value.tableIdentifier);
```

Comment:
I seem to be getting an error on this line, when exploring a table, although the table view works:
```
setCurrentGloabalState called
rejected promise not handled within 1 second: TypeError: Cannot read properties of undefined (reading 'some')
extensionHostProcess.js:147
stack trace: TypeError: Cannot read properties of undefined (reading 'some')
	at f.value (c:\Users\-\Downloads\DSL-test\packages\safe-ds-vscode\dist\extension\mainClient.cjs:79548:46)
	at n._deliver (c:\Users\-\AppData\Local\Programs\VSCodium\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:80:2465)
	at n.fire (c:\Users\-\AppData\Local\Programs\VSCodium\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:80:2784)
	at C.$onMessage (c:\Users\-\AppData\Local\Programs\VSCodium\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:145:36686)
	at u._doInvokeHandler (c:\Users\-\AppData\Local\Programs\VSCodium\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:150:13724)
	at u._invokeHandler (c:\Users\-\AppData\Local\Programs\VSCodium\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:150:13406)
	at u._receiveRequest (c:\Users\-\AppData\Local\Programs\VSCodium\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:150:12173)
	at u._receiveOneMessage (c:\Users\-\AppData\Local\Programs\VSCodium\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:150:11074)
	at f.value (c:\Users\-\AppData\Local\Programs\VSCodium\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:150:8938)
	at n._deliver (c:\Users\-\AppData\Local\Programs\VSCodium\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:80:2465)
	at n.fire (c:\Users\-\AppData\Local\Programs\VSCodium\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:80:2784)
	at o.fire (c:\Users\-\AppData\Local\Programs\VSCodium\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:103:16764)
	at f.value (c:\Users\-\AppData\Local\Programs\VSCodium\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:171:9485)
	at n._deliver (c:\Users\-\AppData\Local\Programs\VSCodium\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:80:2465)
	at n.fire (c:\Users\-\AppData\Local\Programs\VSCodium\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:80:2784)
	at o.fire (c:\Users\-\AppData\Local\Programs\VSCodium\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:103:16764)
	at MessagePortMain.<anonymous> (c:\Users\-\AppData\Local\Programs\VSCodium\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:171:7562)
	at MessagePortMain.emit (node:events:513:28)
	at MessagePortMain._internalPort.emit (node:electron/js2c/utility_init:2:367)
	at Object.callbackTrampoline (node:internal/async_hooks:130:17)
```

This error occurs after loading the table or moving columns around.